### PR TITLE
Fix fromto capsule/cylinder orientation in MJCF parser

### PR DESCRIPTION
**Pull request by adenzler-nvidia**
## Summary

- Fix `fromto` capsule/cylinder orientation: the parser aligned the long axis with Y then rotated Z→Y, but both Newton and MuJoCo shapes extend along Z. This produced wrong geom orientations for robots using `fromto` geoms (e.g. Unitree H1)
- Use `wp.quat_between_vectors` to align Z with `start - end` (matching MuJoCo's `mjuu_z2quat`), with explicit 180-degree fallback for the anti-parallel case
- Remove dead `geom_up_axis` correction that was always identity (Z→Z)
- Add analytical unit test verifying fromto position and orientation for diagonal, downward, and anti-parallel cases

## Test plan

- [x] `TestFromtoCapsuleOrientation` — 3 tests verifying position and Z-axis alignment
- [x] `TestMenagerie_UniversalRobotsUr5e` passes
- [x] `TestMenagerie_ApptronikApollo` passes
- [x] `TestMenagerie_UnitreeG1` passes
- [x] `TestMenagerie_UnitreeH1` passes (was failing on `geom_xmat` before this fix)

## Summary by CodeRabbit

* **Bug Fixes**
  * Corrected capsule and cylinder orientation handling when importing MJCF so shapes align reliably with their defined endpoints, including robust handling of degenerate edge cases.

* **Tests**
  * Added end-to-end tests verifying midpoint positioning and axis alignment for capsules and cylinders across diagonal, upward, and downward configurations.

### CODE REVIEW

File: newton/_src/utils/import_mjcf.py [hunk 1]
```diff
@@ -744,26 +743,24 @@ def parse_shapes(
...
 
-                    # compute rotation to align the Warp capsule (along x-axis), with mjcf fromto direction
-                    axis = wp.normalize(end - start)
-                    angle = math.acos(wp.dot(axis, wp.vec3(0.0, 1.0, 0.0)))
-                    axis = wp.normalize(wp.cross(axis, wp.vec3(0.0, 1.0, 0.0)))
-
+                    # Compute pos and quat matching MuJoCo's fromto convention:
+                    # direction = start - end, align Z axis with it (mjuu_z2quat).
+                    # quat_between_vectors degenerates for anti-parallel vectors,
+                    # so handle that case with an explicit 180° rotation around X.
```
Comment (vreutskyy) [inline newton/_src/utils/import_mjcf.py:749]: If `start == end`, `wp.normalize(start - end)` yields a zero vector and `wp.quat_between_vectors` with a zero input is undefined (NaN/garbage). Consider guarding:

```python
dir_vec = start - end
if wp.length_sq(dir_vec) < 1e-16:
    geom_rot = wp.quat_identity()
else:
    direction = wp.normalize(dir_vec)
    if float(direction[2]) < -0.999999:
        geom_rot = wp.quat(1.0, 0.0, 0.0, 0.0)  # 180° around X
    else:
        geom_rot = wp.quat_between_vectors(wp.vec3(0.0, 0.0, 1.0), direction)
```

File: newton/tests/test_import_mjcf.py [hunk 2]
```diff
@@ -6906,3 +6906,65 @@ def test_include_before_compiler_with_nested_includes(self):
             builder = newton.ModelBuilder()
             builder.add_mjcf(main_path)
             self.assertEqual(builder.body_count, 2)
```
Comment (vreutskyy) [inline newton/tests/test_import_mjcf.py:6908]: The fix applies to both capsule and cylinder (same code path), but only capsules are tested here. Could you add one cylinder fromto case to close that gap?
